src/branchandbound.jl

```diff
@@ -18,7 +18,7 @@ function _branch_bound(ba::BranchAndBoundEnclosure, f::Function, X::Interval_or_
     range_extrema, flag = _monotonicity_check(f, X, dfX)
     flag && return hull(range_extrema, initial)
 
-    fX = f(X...)  # TODO: allow user to choose how to evaluate this (mean value, natural enclosure)
+    fX = f(X)  # TODO: allow user to choose how to evaluate this (mean value, natural enclosure)
```

Comment:
Do we have an issue for this TODO?

Reply:
https://github.com/JuliaReach/RangeEnclosures.jl/issues/77 this was about adding mean value form and allow the user to choose the evaluation mechanism, it was closed in a previous PR, but I reopened since it seems to fit nicely this TODO